pymodbus/transaction.py

```diff
@@ -368,16 +374,13 @@ def _transact(self, request: ModbusRequest, response_length, full=False, broadca
             packet = self.client.framer.buildPacket(request)
             Log.debug("SEND: {}", packet, ":hex")
             size = self._send(packet)
-            if (
-                isinstance(size, bytes)
-                and self.client.state == ModbusTransactionState.RETRYING
-            ):
+            if self.client.state == ModbusTransactionState.RETRYING:
                 Log.debug(
                     "Changing transaction state from "
                     '"RETRYING" to "PROCESSING REPLY"'
                 )
                 self.client.state = ModbusTransactionState.PROCESSING_REPLY
-                return size, None
+                return b"", None
```

Comment:
This is wrong, we only change to REPLY when we have a reply.

This have nothing to do with typing, and it is not covered by tests.

Reply:
> This is wrong, we only change to REPLY when we have a reply.

The issue here is that we can't return an integer, the calling code expects bytes to be returned.

Reply:
The code as it is works or not ?

In any case a test case would show the current code to be faulty, and your new code to be correct.

Reply:
> The code as it is works or not ?

I guess? I mean the tests seem to pass either way so it's a bit hard to tell.

I went ahead and removed this return entirely which also seems to work.